python/lsst/ts/ATHexapod/controller.py

```diff
@@ -0,0 +1,502 @@
+import asyncio
+import logging
+
+
+class ATHexapodController:
+    def __init__(self, log=None):
+
+        self.host = '127.0.0.1'
+        self.port = 50000
+        self.timeout = 2.
+        self.long_timeout = 30.
+
+        self.reader = None
+        self.writer = None
+
+        self.lock = asyncio.Lock()
+
+        self.log = log
+
+        if self.log is None:
+            self.log = logging.getLogger(__name__)
+
+    def is_connected(self):
```

Comment:
You might consider making this a property (with the `@property` decorator). It's a very minor suggestion. I like properties for simple things like this because they feel natural to me and save a bit of typing (`foo.is_connected` instead of `foo.is_connected()`). But it's certainly a matter of taste, and even if you like the idea it may not be worth the trouble to change.